Source/JavaScriptCore/runtime/WeakMapPrototype.cpp

```diff
@@ -213,7 +213,12 @@ JSC_DEFINE_HOST_FUNCTION(protoFuncWeakMapGetOrInsertComputed, (JSGlobalObject* g
                 RETURN_IF_EXCEPTION(scope, { });
             }
 
-            map->addBucket(vm, keyCell, value, hash, index);
+            // FIXME: rdar://145147128
+            // Could we detect if the map is changed and only rehash if needed?
+
+            // Call to valueCallback can modify our state, so we need to re-check the index
+            auto [bucketIndex, _] = map->findBucketIndex(keyCell, hash);
```

Comment:
I don't think we have a specific way to tell whether the weak map is refreshed or not. Maybe we can toggle a flag once we do
```
template<typename WeakMapBucket>
void WeakMapImpl<WeakMapBucket>::rehash(RehashMode mode) { ... }
```

Reply:
Discussed with Yusuke. We should apply a conservative fix first for weak map and leave a FIXME for further optimization.